hr_holidays_overview/models/hr_employee_hour.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2022 Camptocamp SA
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import logging
+
+from odoo import api, fields, models
+
+TYPE_SELECTION = [
+    ("leave_request", "Leave Request"),
+    ("leave_allocation", "Leave Allocation"),
+]
+ONDELETE_SELECTION = {
+    "leave_request": "set default",
+    "leave_allocation": "set default",
+}
+
+_logger = logging.getLogger(__name__)
+
+
+class HrEmployeeHour(models.Model):
+    _inherit = "hr.employee.hour"
+
+    type = fields.Selection(selection_add=TYPE_SELECTION, ondelete=ONDELETE_SELECTION)
+    leave_state = fields.Selection(
+        [
+            ("draft", "To Submit"),
+            ("cancel", "Cancelled"),
+            ("confirm", "To Approve"),
+            ("refuse", "Refused"),
+            ("validate1", "Second Approval"),
+            ("validate", "Approved"),
+        ],
+        "State",
+    )
+
+    @api.model
+    def search_allocations_domain(self, employee):
+        return [
+            ("employee_id", "=", employee.id),
+            ("state", "not in", ("draft", "cancel", "refuse")),
+            ("date_from", ">=", employee.last_hours_report_date),
+        ]
+
+    @api.model
+    def search_requests_domain(self, employee):
+        """ Search filter rules for requests """
+        return [
+            ("employee_id", "=", employee.id),
+            ("state", "not in", ("draft", "cancel", "refuse")),
+            ("date_from", ">=", employee.last_hours_report_date),
+        ]
+
+    @api.model
+    def _prepare_leave_allocation_values(self, employee, date_from=None, date_to=None):
+        """Retrieve leave allocations values
+
+        :param employee: an employee record
+        :param date_from: a datetime.date object (default first contract date, included)
+        :param date_to: a datetime.date object (default today, included)
+        """
+        hla_model = self.env["hr.leave.allocation"]
+        search_domain = self.search_allocations_domain(employee)
+        allocations = hla_model.search(search_domain)
```

Comment:
```suggestion
        allocations = hla_model.with_context(active_test=False).search(
            search_domain
        )
```
We want to manage all records independently of their active state. It's mandatory to be able to do statistics and have a decisional POV with the report.